src/wireguard.rs

```diff
@@ -0,0 +1,305 @@
+ pub struct UpdateKeyPair
+ {
+    pub privkey : [u8; SIZE_PRIVKEY],
+    pub pubkey : [u8; SIZE_PUBKEY],
+    pub label : String
+ }
+
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct DeleteKeyPair
+ {
+     pub uid : u32 // unique ID 
+ }
+
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct GenerateKeyPair
+ {
+    pub uid : u32, // unique ID 
+    pub label : String
+ }
+ 
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct ListKeys
+ {
+    //empty
+ }
+ 
+
+
+ /*
+    Reponses
+ */
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct AEAD(pub [u8;32]);
+
+
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct KeyResponse
+ {
+    pub key_info : KeyInfo
+ }
+
+
+ /**/
+
+ // To be serialized and safed in the trussed store
+ #[derive(Clone, Debug, PartialEq)]
+ #[allow(missing_docs)]
+ pub struct KeyInfo 
+ {
+
+    id : u64,
+    label : String,
+    pubkey : [u8; SIZE_PUBKEY]
+ }
+
+
+// implementations
+
+// hex
+impl core::fmt::Display for AEAD {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{:x?}", self.0)
+    }
+}
+
+impl core::fmt::Display for KeyResponse {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{:x?} w/ UID: {} pubkey : {:x?}", self.key_info.id, self.key_info.label, self.key_info.pubkey)
+    }
+}
+
+impl core::fmt::Display for GetAead {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "\nC: \t\t{:x?}\nH: \t\t{:x?}\nPubkey: \t{:x?}\nkeyid: \t\t{:x?}\n", self.c, self.h, self.pubkey, self.key_id)
+    }
+}
+
+
+#[derive(Clone, Debug, PartialEq)]
+/// The public API of this wireguard m
+#[allow(missing_docs)]
+pub enum WgCommand {
+    Unlock(Unlock),
+    RegisterKeyPair(RegisterKeyPair),
+    DeleteKeyPair(DeleteKeyPair),
+    UpdateKeyPair(UpdateKeyPair),
+    ListKeys(ListKeys),
+
+    GenerateKeyPair(GenerateKeyPair),
+    GetAead(GetAead)
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+/// The `serde::Serialize` and `serde::Deserialize` implementations allow
+/// credentials to easily be stored in binary format.
+pub struct PrivateKey {
+    uid: u64,
+    key_handle: trussed::types::ObjectHandle,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+/// The `serde::Serialize` and `serde::Deserialize` implementations allow
+/// credentials to easily be stored in binary format.
+pub struct Credential {
+    label: trussed::ByteBuf<consts::U256>,
+    period_seconds: u64,
+    key_handle: trussed::types::ObjectHandle,
+}
+
+#[allow(missing_docs)]
+impl<T> Wireguard<T>
+where
+    T: trussed::Client
+{
+   /// Constructor, consumes a Trussed client
+   pub fn new(trussed: T) -> Self {
+    Self { trussed }
+    }
+
+    pub fn unlock(&mut self, parameters: &Unlock) -> Result<()> {
+
+        print!("Unlock called: {:?}", parameters);
```

Comment:
I'd suggest using our `delog` library for this. Possibly needs out-of-band discussion to explain how to use. The point is, it will let you have `debug!`, `info!` etc. calls in the code that get completely compiled out, unless explicitly opted in from the runner.